### PR TITLE
DOC: Automatic flowchart

### PR DESCRIPTION
**Pull request by larsoner**
### Before merging …

- [x] Changelog has been updated (`docs/source/changes.md`)

Locally produces:

Screenshot from 2024-02-23 14-52-05

@hoechenberger feel free to push directly wording changes you want. Locally you can iterate pretty easily with:
```
~/python/mne-bids-pipeline/docs/site$ python -m http.server
~/python/mne-bids-pipeline/docs/site$ ../build_docs.sh
```
then refreshting `http://localhost:8000/features/overview.html` in your browser.

### CODE REVIEW

**Comment by larsoner** [comment]
@SophieHerbst can you see if this fixes the wording issue(s) you ran into or if there is more to do?

File: mne_bids_pipeline/steps/preprocessing/_08a_apply_ica.py [hunk 1]
```diff
@@ -224,7 +224,7 @@ def apply_ica_raw(
     raw_fname = in_files.pop(in_key)
     assert len(in_files) == 0, in_files
     out_files = dict()
-    out_files[in_key] = raw_fname.copy().update(processing="clean")
+    out_files[in_key] = raw_fname.copy().update(processing="clean", split=None)
```
Comment (hoechenberger) [inline mne_bids_pipeline/steps/preprocessing/_08a_apply_ica.py:227]: Should we check **all** code that generates output filenames to ensure we set `split=None` there?

Comment (larsoner) [reply mne_bids_pipeline/steps/preprocessing/_08a_apply_ica.py:227]: I think we could do it in the function that prepares the output filenames actually. I'll give it a shot

Comment (larsoner) [reply mne_bids_pipeline/steps/preprocessing/_08a_apply_ica.py:227]: Should be working now!

**Comment by hoechenberger** [comment]
most of the steps are optional though.... we'd end up with almost everything being marked as such

**Comment by SophieHerbst** [comment]
mh right. I am wondering how to make it clearer which config options trigger which steps (or their re-run)

**Comment by hoechenberger** [comment]
there are `get_config()` functions in every step now which cover almost all configuration options that affect this step (exceptions are I believe n_jobs, subjects and session, but i'm not 100% certain from the top of my head)

in theory this information could be used

**Comment by SophieHerbst** [comment]
Used as in informing the documentation? 
(I learned about this part last week, when I forgot to mention the image_kwargs for the report there; easy to make mistakes as it needs to be added manually)

**Comment by larsoner** [comment]
Opened #862 and #861 so we can iterate further in follow-up PRs. This one in the meantime is 90% a bugfix for missing / incorrect information at the moment (the other 10% being the ICA/SSP enhancement to try and make that part clearer) -- @SophieHerbst @hoechenberger happy with those parts so we can merge this?

**Comment by larsoner** [comment]
https://output.circle-artifacts.com/output/job/d69d1de1-f4fc-4714-9d6e-f43e8f3816b7/artifacts/0/site/features/overview.html

**Comment by hoechenberger** [comment]
yes, this is great! feel free to merge whenever you feel it's ready 👍👍👍

**Review by hoechenberger — APPROVED** [review]

**Pull request merged**